pandas/core/dtypes/cast.py

```diff
@@ -564,6 +564,9 @@ def invalidate_string_dtypes(dtype_set):
 
 def coerce_indexer_dtype(indexer, categories):
     """ coerce the indexer input array to the smallest dtype possible """
```

Comment:
best to add a doc-string here; why are you adding these changes?

Reply:
Done.